frontend/src/components/PositiveInput.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react'
+import { Input } from '../styling/styles'
+
+interface Props extends React.InputHTMLAttributes<HTMLInputElement> {
+  precision?: number
+}
+
+export const PositiveInput: React.FC<Props> = (props) => {
+  const { onChange, precision } = props
+
+  const respectsPrecision = (value: string) => {
+    if (precision !== undefined) {
+      return value.split('.')[1].length <= precision
+    }
+    return true
+  }
+
+  const positiveOnChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const pattern = new RegExp(`(^\\d+$|^\\d+\\.\\d+$|^\\d+\\.$|^$)`)
```

Comment:
can pass in the pattern directly as one of `Input`'s props

Reply:
When I tried passing the pattern as a prop, it would still allow users to input invalid strings. This is probably due to the fact that the [pattern attribute is only used for form submissions](https://www.w3schools.com/tags/att_input_pattern.asp). The current approach would disallow completely inputting strings that do not match the pattern

Reply:
Ahhh! thats subtle but v important! thanks for clarifying! ignore the change comment